### ISSUE TITLE
Failed to listen on IPv4

### ISSUE THREAD

**Issue by ghost** [lstrojny/homebridge-prometheus-exporter#22]
**Describe The Bug:**
The plugin doesn't listen to the given port. 

The code currently uses `::` as the host to listen. This should probably be changed to `localhost` or provide a config option. https://www.fastify.io/docs/latest/Reference/Server/#listen 
 
https://github.com/lstrojny/homebridge-prometheus-exporter/blob/57e8a4d831820065b20670af3a4d6233ed5ff7f0/src/adapters/http/fastify.ts#L61

**To Reproduce:**
I have IPv6 disabled on my Raspberry Pi. The plugin won't start cleanly for me in this case. 

**Expected behavior:**
It should be able to listen on IPv4 as well. 

**Logs:**

```
[16/11/2022, 09:51:30] [PrometheusExporter] Failed to start Prometheus HTTP server on port 36123: Error: listen EAFNOSUPPORT: address family not supported :::36123
    at Server.setupListenHandle [as _listen2] (node:net:1355:21)
    at listenInCluster (node:net:1420:12)
    at doListen (node:net:1559:7)
    at processTicksAndRejections (node:internal/process/task_queues:84:21) {
  [stack]: 'Error: listen EAFNOSUPPORT: address family not supported :::36123\n' +
    '    at Server.setupListenHandle [as _listen2] (node:net:1355:21)\n' +
    '    at listenInCluster (node:net:1420:12)\n' +
    '    at doListen (node:net:1559:7)\n' +
    '    at processTicksAndRejections (node:internal/process/task_queues:84:21)',
  [message]: 'listen EAFNOSUPPORT: address family not supported :::36123',
  code: 'EAFNOSUPPORT',
  errno: -97,
  syscall: 'listen',
  address: '::',
  port: 36123
}
```

**Plugin Config:**

```json
{
    "pin": "548-70-314",
    "debug": true,
    "prefix": "homebridge",
    "port": 36123,
    "refresh_interval": 60,
    "request_timeout": 10,
    "discovery_timeout": 20,
    "platform": "PrometheusExporter"
}
```

**Screenshots:**

**Environment:**

* **Plugin Version**: v0.0.10
* **Homebridge Version**: 1.5.1
* **Node.js Version**: v16.15.0
* **NPM Version**: 8.5.5
* **Operating System**: Raspbian v1.0.24

**Comment by lstrojny** [comment]
Interesting, according to https://www.fastify.io/docs/latest/Reference/Server/#listen, "::" should do work both for IPv4 and IPv6 but it it doesn’t work as advertised apparently. Offering a configuration option is probably the right thing™️ to do.

**Comment by ghost** [comment]
There is a note at the bottom of that section that it is dependent on OS. So it is not guaranteed always I believe 

> * Using :: for the address will listen on all IPv6 addresses and, depending on OS, may also listen on [all IPv4 addresses](https://nodejs.org/api/net.html#serverlistenport-host-backlog-callback).

**Comment by lstrojny** [comment]
Out of curiosity, could you try and edit `/usr/lib/node_modules/homebridge-prometheus-exporter/dist/src/adapters/http/fastify.js` on your host and replace `::` with `0.0.0.0`?

**Comment by lstrojny** [comment]
Version 0.0.11 has been released which includes a fallback fix for the issue. Adding the configuration option is still something I should probably do.

**Comment by lstrojny** [comment]
Correction, 0.0.12 is the version that includes the fix. 0.0.11 has been pulled, I hosed the release

**Comment by ghost** [comment]
That worked. thanks.

**Comment by lstrojny** [comment]
Great, thanks for the feedback. So we have a solution now to make it work as expected for probably 80% of the user base. Next step is to add the configuration option for the interface to bind to.

**Comment by lstrojny** [comment]
Option `interface` added in `develop` branch. Will be included in the next release, 0.0.13
